### PR TITLE
[WFLY-16972] Rewriting DiscoveryService to use new MSC value API

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.redhat.com/browse/WFLY-16972

### CODE REVIEW

File: ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/DiscoveryService.java [hunk 1]
```diff
@@ -22,45 +22,89 @@
...
+        discoveryConsumer.accept(null);
+        synchronized (providers) {
+            if (remotingDP != null) {
+                providers.remove(remotingDP);
+                remotingDP = null;
+            }
+            if (assocationDP != null) {
+                providers.remove(assocationDP);
+                assocationDP = null;
+            }
```
Comment (chengfang) [inline ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/DiscoveryService.java:96]: do we also need to remove any extra discovery providers added via line 105?

**Comment by chengfang** [comment]
@rachmatowicz fyi

File: ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/DiscoveryService.java (hunk 1 above)
Comment (ropalka) [reply ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/DiscoveryService.java:96]: Looking at the code I don't think so @chengfang - these are registered via DiscoveryRegistrationProcessor (lines 73 & 74)
DiscoveryService can be restarted any time so everything done in start() method must be undone in stop() method.

Comment (ropalka) [reply ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/DiscoveryService.java:96]: To be more precise @chengfang this PR doesn't change anything from functional point of view.
It is just one refactoring step that only refactors dependency configuration staff to use a new MSC value API.
Notice that both external methods (in DiscoveryRegistrationProcessor code):
 * HttpDiscoveryConfigurator.configure() 
 * RemoteEJBDiscoveryConfigurator.configure()

are just injecting new provider into DiscoveryService.
This is what was happening before this PR and this what will be happening after this PR (if merged).

**Comment by ropalka** [comment]
Any update WRT this PR @chengfang ?

**Review by chengfang — APPROVED** [review]

**Comment by bstansberry** [comment]
@chengfang This isn't in the 27 Beta1 tag. Please apply either the 28.x or 27.x label to this and update the Fix Version on the JIRA to reflect whether this should go into the Oct 12/13 27 Final, or wait for 28. Whatever you decide is best is ok with me. Thanks!

**Comment by bstansberry** [comment]
Thanks @ropalka

**Pull request merged**